thirtyinch/src/main/java/net/grandcentrix/thirtyinch/TiPresenter.java

```diff
@@ -155,12 +151,37 @@ public void bindNewView(@NonNull final V view) {
                             + "Binding a view is not allowed");
         }
 
-        if (view == null) {
-            throw new IllegalStateException(
-                    "the view cannot be set to null. Call #sleep() instead");
+        if (isViewAttached()) {
+            if (view.equals(mView)) {
+                TiLog.v(TAG, "not calling onAttachView(), view already attached");
+                return;
+            } else {
+                throw new IllegalStateException(
+                        "a view is already attached, call #detachView first");
+            }
+        }
+
+        if (!isInitialized()) {
+            throw new IllegalStateException("Presenter is not created, call #create() first");
         }
 
         mView = view;
+        moveToState(State.VIEW_ATTACHED, false);
+        mCalled = false;
```

Comment:
mCalled isn't a good name.
I think it would be nicer if we have two booleans. `mViewAttachedCalled` and `mViewDetachedCalled`.
So we can't override the single boolean anymore...